jupyter_client/kernelspecapp.py

```diff
@@ -19,14 +20,32 @@
 class ListKernelSpecs(JupyterApp):
     description = """List installed kernel specifications."""
     kernel_spec_manager = Instance(KernelSpecManager)
+    json_output = Bool(False, help='output spec name and location as machine-readable json.',
+            config=True)
+    
+    flags = {'json': ({'ListKernelSpecs': {'json_output': True}},
+                "output spec name and location as machine-readable json."),
+             'debug': base_flags['debug'],
+            }
 
     def _kernel_spec_manager_default(self):
         return KernelSpecManager(parent=self, data_dir=self.data_dir)
 
     def start(self):
-        print("Available kernels:")
-        for kernelname in sorted(self.kernel_spec_manager.find_kernel_specs()):
-            print("  %s" % kernelname)
+        specs = sorted(self.kernel_spec_manager.find_kernel_specs().items())
+        if not self.json_output:
+            print("Available kernels:")
+            for kernelname, spec in specs:
+                print("  %s" % kernelname)
+        else:
+            print(json.dumps({
+                "kernelspecs":
+                    [{'name':k, 'path':v} for k,v in specs]
```

Comment:
Should this also include the info from the kernelspecs? In the particular case of the native kernelspec, the `kernel.json` file may not actually be present in the resource directory, because that dictionary is generated without writing it somewhere. Alternatively, we could do something like writing it to a temporary directory when we look it up.